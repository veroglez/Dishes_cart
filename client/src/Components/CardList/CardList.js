import React, { PureComponent } from 'react';
import { arrayOf, shape } from 'prop-types';
import { Card } from '..';
import style from './CardList.module.scss';

class CardList extends PureComponent {
  static propTypes = {
    dishes: arrayOf(shape({})).isRequired,
  };

  render() {
    const { props: { dishes } } = this;

    return (
      <div className={style.cardList}>
        {dishes.map((dish, index) => (
          <Card
            key={index.toString()}
            index={index}
            image={dish.image}
            title={dish.name}
            price={dish.price}
          />
        ))}
      </div>
    );
  }
}

export default CardList;
