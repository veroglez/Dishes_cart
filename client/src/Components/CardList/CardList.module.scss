.cardList {
  margin: 90px 40px;
  display: flex;
  flex-wrap: wrap;
}