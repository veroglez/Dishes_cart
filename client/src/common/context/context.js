import { node } from 'prop-types';
import React, { createContext, Component } from 'react';
import l10n from '../l10n';

const Context = createContext();
const { Provider, Consumer: ConsumerData } = Context;

class ProviderData extends Component {
  static propTypes = {
    children: node,
  };

  static defaultProps = {
    children: undefined,
  };

  state = {
    l10n,
    products: [],
    items: 0,
    total: 0,
  }

  onData = object => this.setState(object);

  render() {
    const {
      onData, props: { children },
    } = this;

    return (
      <Provider value={{ ...this.state, onData }}>
        { children }
      </Provider>
    );
  }
}

export { ConsumerData, ProviderData };