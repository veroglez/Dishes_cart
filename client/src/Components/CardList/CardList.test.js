import React from 'react';
import renderer from 'react-test-renderer';
import CardList from './CardList';
import { ProviderData } from '../../common/context/context';

const DISHES = [
  { name: 'lorem ipsum 1', image: '/image.png', price: 20 },
  { name: 'lorem ipsum 2', image: '/image.png', price: 30 },
];
const DEFAULTS = {
  l10n: require('../../common/l10n'), // eslint-disable-line
};

const HOC = inherit => (
  <ProviderData>
    <CardList {...DEFAULTS} {...inherit} />
  </ProviderData>
);

describe('<CardList>', () => {
  it('when {dishes}', () => {
    const tree = renderer.create(<HOC dishes={DISHES} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
