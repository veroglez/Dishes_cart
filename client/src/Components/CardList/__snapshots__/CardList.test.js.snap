// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardList> when {dishes} 1`] = `
<div
  className="cardList"
>
  <div
    className="card"
  >
    <img
      alt="lorem ipsum 1"
      className="image"
      src="/image.png"
    />
    <div
      className="info"
    >
      <p
        className="title"
      >
        lorem ipsum 1
      </p>
      <p
        className="price"
      >
        0.2 €
      </p>
      <button
        onClick={[Function]}
        type="button"
      >
        Añadir
      </button>
    </div>
  </div>
  <div
    className="card"
  >
    <img
      alt="lorem ipsum 2"
      className="image"
      src="/image.png"
    />
    <div
      className="info"
    >
      <p
        className="title"
      >
        lorem ipsum 2
      </p>
      <p
        className="price"
      >
        0.3 €
      </p>
      <button
        onClick={[Function]}
        type="button"
      >
        Añadir
      </button>
    </div>
  </div>
</div>
`;
